Tracker/Scenes/Trackers Flow/Category settings/CategorySettingsViewController.swift

```diff
@@ -0,0 +1,176 @@
+
+import UIKit
+
+protocol CategorySettingsViewControllerProtocol {
+    func categorySettingsDidUpdated(categories: [TrackerCategory])
+}
+
+final class CategorySettingsViewController: UIViewController {
+    // MARK: - Properties
+    private let coordinator: SettingsFlowCoordinator
+    private var category: TrackerCategory?
+    private let categories: [TrackerCategory]
+    private let delegate: CategorySettingsViewControllerProtocol
+
+    private var currentCategory: TrackerCategory?
+
+    private lazy var nameTextField: UITextField = {
+        let textField = UITextField()
+        textField.textColor = .Custom.text
+        textField.font = UIFont.systemFont(ofSize: 17)
+        textField.backgroundColor = .Custom.actionBackground
+        textField.placeholder = "Введите название категории"
+        textField.layer.cornerRadius = 16
+        textField.clearButtonMode = .whileEditing
+        textField.leftViewMode = UITextField.ViewMode.always
+        textField.addTarget(self, action: #selector(textFieldDidChange(_:)), for: .editingChanged)
+        textField.delegate = self
+        textField.leftView = UIView(frame:CGRect(x:0, y:0, width:10, height:textField.frame.height))
+        textField.translatesAutoresizingMaskIntoConstraints = false
+        return textField
+    }()
+
+    private let warningLabel: UILabel = {
+        let label = UILabel()
+        label.text = ""
+        label.font = UIFont.systemFont(ofSize: 17)
+        label.textColor = .Custom.red
+        label.textAlignment = .center
+        label.translatesAutoresizingMaskIntoConstraints = false
+        return label
+    }()
+
+    private let saveButton = CustomButton(title: "Готово")
+
+    // MARK: - Initialiser
+    init(coordinator: SettingsFlowCoordinator, edit category: TrackerCategory?, in categories: [TrackerCategory], delegate: CategorySettingsViewControllerProtocol) {
+        self.coordinator = coordinator
+        self.category = category
+        self.categories = categories
+        self.delegate = delegate
+        super.init(nibName: nil, bundle: nil)
+    }
+
+    required init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+
+    // MARK: - LifeCycle
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        view.backgroundColor = .white
+        navigationItem.hidesBackButton = true
+        taps()
+        layout()
+        hideKeyboardWhenTappedAround()
+    }
+
+    override func viewWillAppear(_ animated: Bool) {
+        super.viewWillAppear(animated)
+        setupView()
+    }
+
+    // MARK: - Methods
+    private func taps() {
+        saveButton.tapAction = { [weak self] in
+            guard let self = self else { return }
+
+            let newTitle = self.nameTextField.text ?? ""
+            let newCategories = self.updateCategories(with: newTitle)
+            self.delegate.categorySettingsDidUpdated(categories: newCategories)
+            self.coordinator.pop()
+        }
```

Comment:
Новый метод крут, хотя я наверное завидую что в проде такое не использую, поэтому побубню такой рекомендацией: 

Лучше выносить код замыкания в отедельный метод. Так читаемее, легче будет дебажить брекпоинтами, да и селф можно опускать
```suggestion
        saveButton.tapAction = { [weak self] in
            self?.onSaveAction()
        }

     }
     
     private func onSaveAction() {
            let newTitle = nameTextField.text ?? ""
            let newCategories = updateCategories(with: newTitle)
            delegate.categorySettingsDidUpdated(categories: newCategories)
            coordinator.pop()
```

Reply:
не дочитала к тому моменту – думала что это апишка кнопки, а это твоя обертка :)

Reply:
✔